great_expectations/data_context/store/datasource_store.py

```diff
@@ -320,3 +322,30 @@ def _determine_datasource_key(self, datasource_name: str) -> DataContextVariable
             resource_name=datasource_name,
         )
         return datasource_key
+
+    def add_batch_config(
+        self, data_asset: FluentDataAsset, batch_config: BatchConfig
+    ) -> BatchConfig:
+        key = DataContextVariableKey(resource_name=data_asset.datasource.name)
+
+        loaded_datasource = self.get(key)
+        assert isinstance(loaded_datasource, FluentDatasource)
+
+        loaded_asset = loaded_datasource.get_asset(data_asset.name)
+        batch_config_names = {bc.name for bc in loaded_asset.batch_configs}
+
+        if batch_config.name in batch_config_names:
+            raise ValueError(
+                f'"{batch_config.name}" already exists (all existing batch_config names are {", ".join(batch_config_names)})'
+            )
+
+        # This MUST be reassigned for it to be picked up during serialization
```

Comment:
The alternative here is to use `_update__fields_set__on_truthyness(self, _BATCH_CONFIGS_KEY)` during serialization, but that feels like more of a smell to me.

Reply:
I think you should be able to do this in order for this field to show up in the serialization with our default of `exclude_unset=True`
```python
self.__fields_set__.add("batch_configs")
```
https://github.com/great-expectations/great_expectations/blob/cb05dbb71f2a968550bed84f1adb2a31956d45bf/great_expectations/datasource/fluent/fluent_base_model.py#L57-L60

Reply:
Thanks! For anyone who stumbles across this, both options work for this PR, but when we get to the corresponding delete method, we'll probably want to be modifying `__fields_set__` directly, so I'm moving this to follow that pattern.